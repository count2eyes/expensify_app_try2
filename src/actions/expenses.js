import React from "react";
import uuid from "uuid";

export const addExpense = ({ description, amount, note, createdAt }) => {
  return {
    type: "ADD_EXPENSE",
    expense: {
      description: description || "",
      amount: amount || 0,
      note: note || "Notes!(not like you didn't notice)",
      createdAt: createdAt || 0,
      id: uuid()
    }
  };
};

export const removeExpense = id => {
  return {
    type: "REMOVE_EXPENSE",
    id
  };
};

export const editExpense = (id, { description, amount, note, createdAt }) => {
  return {
    type: "EDIT_EXPENSE",
    updates: { description, amount, note, createdAt },
    id
  };
};
