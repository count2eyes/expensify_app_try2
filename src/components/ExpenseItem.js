import React from "react";
import moment from "moment";
import { Link } from "react-router-dom";
import numeral from "numeral";

const ExpenseItem = props => {
  const { description, amount, note, createdAt, id } = props.expense;
  return (
    <div>
      <Link to={`/edit/${id}`}>
        <h3>{description}</h3>
      </Link>
      <p>
        {numeral(amount / 100).format("$0,0.00")} -{" "}
        {moment(createdAt).format("ddd DD MMM, YYYY")} <small>{note}</small>
      </p>
    </div>
  );
};

export default ExpenseItem;
