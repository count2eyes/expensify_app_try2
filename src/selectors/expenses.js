import moment from "moment";

const selector = (expenses, filters) => {
  return expenses
    .filter(expense => {
      const matchedText = filters
        ? expense.description.includes(filters.text)
        : true;
      const matchedStartDate = filters.startDate
        ? moment(expense.createdAt).isSameOrAfter(filters.startDate, "day")
        : true;
      const matchedEndDate = filters.endDate
        ? moment(expense.createdAt).isSameOrBefore(filters.endDate, "day")
        : true;

      return matchedText && matchedStartDate && matchedEndDate;
    })
    .sort((a, b) => {
      if (filters.sortBy === "date") {
        return b.createdAt - a.createdAt;
      } else if (filters.sortBy === "amount") {
        return b.amount - a.amount;
      }
    });
};

export default selector;
