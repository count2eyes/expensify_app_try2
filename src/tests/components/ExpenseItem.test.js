import React from "react";
import { shallow } from "enzyme";
import ExpenseItem from "../../components/ExpenseItem";
import expenses from "../fixtures/expenses";

test("should render ExpenseItem correctly", () => {
  const wrapper = shallow(<ExpenseItem expense={expenses[2]} />);
  expect(wrapper).toMatchSnapshot();
});
