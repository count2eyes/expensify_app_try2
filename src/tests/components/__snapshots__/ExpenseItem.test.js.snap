// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseItem correctly 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/3"
  >
    <h3>
      dinner
    </h3>
  </Link>
  <p>
    $0.24
     -
     
    Tue 30 Dec, 1969
     
    <small>
      meh
    </small>
  </p>
</div>
`;
